frontend/src/components/pages/VolunteerDashboard/ShiftCard.tsx

```diff
@@ -0,0 +1,129 @@
+import { Box, Button, Stack, Text } from "@chakra-ui/react";
+import { format, parse } from "date-fns";
+import React, { useEffect, useState } from "react";
+import { useHistory } from "react-router-dom";
+
+import DonorAPIClient from "../../../APIClients/DonorAPIClient";
+import * as Routes from "../../../constants/Routes";
+import {
+  CheckInWithShiftType,
+  ScheduleWithShiftType,
+  ShiftType,
+} from "../../../types/VolunteerTypes";
+import CardField from "../../common/CardField";
+import { getShiftColor } from "./types";
+
+interface CheckInOrScheduleProps {
+  donorId?: string;
+  isPickup?: boolean;
+  pickupLocation?: string;
+  volunteerTime?: string;
+  startTime?: string;
+  startDate?: string;
+  notes: string;
+  type: ShiftType;
+}
+const VolunteerShiftCard = ({
+  shift,
+}: {
+  shift: CheckInWithShiftType | ScheduleWithShiftType;
+}): JSX.Element => {
+  const {
+    donorId,
+    isPickup,
+    pickupLocation,
+    volunteerTime,
+    startTime,
+    startDate,
+    notes,
+    type,
+  } = shift as CheckInOrScheduleProps;
+  const [businessName, setBusinessName] = useState<string>("");
+
+  const history = useHistory();
+
+  const dateLocal = () => {
+    if (startDate) {
+      return format(new Date(startDate), "EEE MMM dd, yyyy");
+    }
+    return startTime && format(new Date(startTime), "EEE MMM dd, yyyy");
+  };
+
+  const timeLocal = () => {
+    if (startDate) {
+      return format(new Date(startDate), "h:mm aa");
+    }
+
+    return (
+      volunteerTime &&
+      format(parse(volunteerTime, "kk:mm", new Date()), "h:mm aa")
+    );
+  };
+
+  useEffect(() => {
+    const getBusinessName = async () => {
+      if (donorId) {
+        const donor = await DonorAPIClient.getDonorById(donorId);
+        setBusinessName(donor.businessName);
+      }
+    };
+
+    getBusinessName();
+  }, []);
+
+  return (
+    <Box my="24px" width="100%" overflow="hidden">
+      <Stack
+        direction={["column", "row"]}
+        display="flex"
+        justify="space-between"
+      >
+        <Text textStyle="mobileHeader4" whiteSpace="nowrap" minWidth="225px">
+          {`${dateLocal()}`}
+          <Text minWidth="125px" textStyle="mobileSmall" color="hubbard.100">
+            {type === ShiftType.CHECKIN && "Fridge check-in"}
+            {type === ShiftType.SCHEDULING &&
+              (isPickup ? "Pickup assistance" : "Unloading assistance")}
+          </Text>
+        </Text>
+        <Button
+          float="right"
+          mt="1.5rem"
+          size="lg"
+          width="35%"
+          variant="viewDetails"
+          onClick={() => history.push(Routes.SCHEDULING_PAGE)}
```

Comment:
let's not have it route anywhere for now, and add it back to use the same component when 
https://www.notion.so/uwblueprintexecs/Volunteer-Checkin-Shift-Signup-Flow-fe45029d2a904a03ba913b40d3bf34eb
and 
https://www.notion.so/uwblueprintexecs/Volunteer-Food-Rescue-Shift-Signup-Flow-6147bb4ee0fc40af809aace5bf72355e
are merged (@carissa-tang please add a ticket to do this)
```suggestion
```